Src/BootCamp.Chapter/BalanceParser.cs

```diff
@@ -0,0 +1,146 @@
+            }
+
+            var person = BalanceStats.FindHighestBalanceEver(contents);
+            
+           
+
+            StringBuilder sb = ConvertToStringBuilder(person);
+
+            var answer = $"{sb.ToString()} had the most money ever. ¤{person[0].HighestBalance()}.";
+            return answer;
+        }
+
+        public static string FindPoorestPerson(string[] contents)
+        {
+            if (IsInValidInput(contents))
+            {
+                return InValidOutput;
+            }
+
+            var person = BalanceStats.FindMostPoorPerson(contents); 
+            StringBuilder sb = ConvertToStringBuilder(person);
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+            NumberFormatInfo noParens = (NumberFormatInfo)CultureInfo.CurrentCulture.NumberFormat.Clone();
+            noParens.CurrencyNegativePattern = 1;
+            var lowestBalanceString = person[0].CurrentBalance().ToString("C0", noParens);
+
+
+            if (sb.ToString().Contains(','))
+            {
+                return $"{sb.ToString()} have the least money. {lowestBalanceString}.";
+            }
+            else
+            {
+                return $"{sb.ToString()} has the least money. {lowestBalanceString}.";
+            }
+
+           
+        }
+
+        public static string FindPersonWithBiggestLoss(string[] contents)
+        {
+            if (IsInValidInput(contents))
+            {
+                return InValidOutput;
+            }
+
+            var person = BalanceStats.FindPersonWithBiggestLoss(contents); 
+            StringBuilder sb = ConvertToStringBuilder(person);
+
+            if (person.Length == 0 )
+            {
+                return InValidOutput;
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+            NumberFormatInfo noParens = (NumberFormatInfo)CultureInfo.CurrentCulture.NumberFormat.Clone();
+            noParens.CurrencyNegativePattern = 1;
+            var lowestBalanceString = person[0].BiggestLoss().ToString("C0", noParens);
+
+            var answer =  $"{sb.ToString()} lost the most money. {lowestBalanceString}.";
+            return answer; 
+           
+        }
+
+
+
+        public static StringBuilder ConvertToStringBuilder(Person[] person)
+        {
+            var sb = new StringBuilder();
+            for (int i = 0; i < person.Length; i++)
+            {
+                if (i == 0)
+                {
+                    sb.Append(person[i].GetName());
+                }
+                else
+                {
+                    sb.Append(", " + person[i].GetName());
+                }
+            }
+
+            if (sb.ToString().Contains(','))
+            {
+                ReplaceCommaWithAnd(sb);
+            }
+
+            return sb;
+        }
+
+        private static void ReplaceCommaWithAnd(StringBuilder message)
+        {
+            var index = message.ToString().LastIndexOf(", ");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Define the locale to be used in this string operation.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5053435)